### DISCUSSION TITLE
brew upgrade --greedy does not upgrade cask

### DISCUSSION THREAD

**Discussion by CodeAlanDebug** [Homebrew/discussions#3841] [Casks]
Hi, 

After I run `brew upgrade --greedy` all my casks have been updated but Google-Chrome-Dev is not upgraded.

Old version: 93.0.4577.8   newest version: 108.0.5359.19

However, it does auto_update inside chrome itself. So I am wondering what happened in between, so I tried to `brew reinstall google-chrome-dev` which should install the latest version of google-chrome-dev but it still install the old version 93 which is the one that is not upgraded. 

I have tried to remove the google-chrome-dev install cache and homebrew downloaded cache but it still install the old version I have every time I tried to install it.

**Comment by SMillerDev** [comment]
> so I tried to brew reinstall google-chrome-dev which should install the latest version of google-chrome-dev

No, it'll reinstall what you have. `brew uninstall google-chrome-dev && brew install google-chrome-dev` will

**Reply by CodeAlanDebug** [reply]
I have also tried this one but it does not install the latest version, still install the old version.

**Reply by SMillerDev** [reply]
What does `brew info google-chrome-dev` say?

**Reply by CodeAlanDebug** [reply]
`brew info google-chrome-dev   `                                                                                                                                                              
==> google-chrome-dev: 108.0.5359.19 (auto_updates)
https://www.google.com/chrome/dev/
/usr/local/Caskroom/google-chrome-dev/108.0.5359.19 (61.7KB)
From: https://github.com/Homebrew/homebrew-cask-versions/blob/HEAD/Casks/google-chrome-dev.rb
==> Name
Google Chrome Dev
==> Description
Web browser
==> Artifacts
Google Chrome Dev.app (App)

**Reply by SMillerDev** [reply]
That does look like it installed the new version. What makes you think it didn't?

**Reply by CodeAlanDebug** [reply]
When I reopen chrome-dev, it shows that chrome is out of date and give me the red update icon, so I checked it in setting, then it shows me that I am on version 93 instead of 108
<img width="1467" alt="Screenshot 2022-10-27 at 18 29 01" src="https://user-images.githubusercontent.com/106265297/198346935-d9ccfe89-903e-4f3c-bbd7-348b738e2a8d.png">

**Reply by SMillerDev** [reply]
What does `ls -al /Applications/` say about Chrome Dev?

**Reply by CodeAlanDebug** [reply]
`ls -al /Applications/ | grep Chrome    `                                                                                                                                          
drwxr-xr-x@   3 alan  staff     96 Jul 21  2021 Google Chrome Dev.app
drwxr-xr-x@   3 alan  staff     96 Oct 20 03:41 Google Chrome.app

I think it it not updated if you see the date

**Reply by SMillerDev** [reply]
It doesn't seem so be linked either. What is the output of the uninstall and install commands?

**Reply by CodeAlanDebug** [reply]
The output for uninstall 

`brew uninstall google-chrome-dev`
==> Uninstalling Cask google-chrome-dev
==> Backing App 'Google Chrome Dev.app' up to '/usr/local/Caskroom/google-chrome-dev/108.0.5359.19/Google Chrome Dev.app'
==> Removing App '/Applications/Google Chrome Dev.app'
==> Purging files for version 108.0.5359.19 of Cask google-chrome-dev


The output for install

`brew install google-chrome-dev`
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 3 taps (homebrew/cask-versions, homebrew/core and homebrew/cask).

You have 1 outdated cask installed.
You can upgrade it with brew upgrade
or list it with brew outdated.

==> Downloading https://dl.google.com/chrome/mac/dev/googlechromedev.dmg
Already downloaded: /Users/alan/Library/Caches/Homebrew/downloads/ea827c9aca74cbb2cbf70ba04d758e05af2178e376c5ae1a15bf1dbe43a87959--googlechromedev.dmg
Warning: No checksum defined for cask 'google-chrome-dev', skipping verification.
==> Installing Cask google-chrome-dev
==> Moving App 'Google Chrome Dev.app' to '/Applications/Google Chrome Dev.app'
🍺  google-chrome-dev was successfully installed!

**Reply by gromgit** [reply]
>```
> ==> Downloading https://dl.google.com/chrome/mac/dev/googlechromedev.dmg
> Already downloaded: /Users/alan/Library/Caches/Homebrew/downloads/ea827c9aca74cbb2cbf70ba04d758e05af2178e376c5ae1a15bf1dbe43a87959--googlechromedev.dmg
>```

I'd bet that's the old DMG from when you first installed v93. Delete that file and `brew install` again.

**Reply by CodeAlanDebug** [reply]
`brew uninstall google-chrome-dev && brew install google-chrome-dev` 
                                                                                                                         
```
==> Uninstalling Cask google-chrome-dev
==> Backing App 'Google Chrome Dev.app' up to '/usr/local/Caskroom/google-chrome-dev/108.0.5359.19/Google Chrome Dev.app'
==> Removing App '/Applications/Google Chrome Dev.app'
==> Purging files for version 108.0.5359.19 of Cask google-chrome-dev
Running `brew update --auto-update`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).

You have 2 outdated formulae and 1 outdated cask installed.
You can upgrade them with brew upgrade
or list them with brew outdated.

==> Downloading https://dl.google.com/chrome/mac/dev/googlechromedev.dmg
######################################################################## 100.0%
Warning: No checksum defined for cask 'google-chrome-dev', skipping verification.
==> Installing Cask google-chrome-dev
==> Moving App 'Google Chrome Dev.app' to '/Applications/Google Chrome Dev.app'
🍺  google-chrome-dev was successfully installed!
```

I tried to do this before but I tried to clean that cache `/Users/alan/Library/Caches/Homebrew/downloads/` again, still on version 93.

**Reply by gromgit** [reply]
_Mea culpa_, I just did a direct download of the above URL, opened the DMG, did _Get Info_ on the app therein...and it really is v93, published 2021-Jul-21.

Perhaps Google reverted that release for some reason.

**Reply by Wavone** [reply]
I tried to download that DMG file, I also get version 93, probably something wrong with the source of google because normal chrome which is the production version is 107 already.

**Comment by cho-m** [comment]
It's possible that the non-universal (x86_64-only) variant is no longer updated or the download URL changed. The other Chrome casks (canary, beta, and stable) all use universal.

Right now, only ARM Macs install the universal (x86_64 + ARM64) variant for `google-chrome-dev`. That installation should have the correct version number:
```console
❯ grep -A1 Version "/Volumes/Google Chrome Dev/Google Chrome Dev.app/Contents/Info.plist"
	<key>CFBundleInfoDictionaryVersion</key>
	<string>6.0</string>
--
	<key>CFBundleShortVersionString</key>
	<string>109.0.5384.0</string>
--
	<key>CFBundleVersion</key>
	<string>5384.0</string>
--
	<key>KSVersion</key>
	<string>109.0.5384.0</string>
--
	<key>LSMinimumSystemVersion</key>
	<string>10.13</string>

❯ file "/Volumes/Google Chrome Dev/Google Chrome Dev.app/Contents/MacOS/Google Chrome Dev"
/Volumes/Google Chrome Dev/Google Chrome Dev.app/Contents/MacOS/Google Chrome Dev: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64:Mach-O 64-bit executable arm64]
/Volumes/Google Chrome Dev/Google Chrome Dev.app/Contents/MacOS/Google Chrome Dev (for architecture x86_64):	Mach-O 64-bit executable x86_64
/Volumes/Google Chrome Dev/Google Chrome Dev.app/Contents/MacOS/Google Chrome Dev (for architecture arm64):	Mach-O 64-bit executable arm64
```
